spring-core/src/main/java/org/springframework/util/StreamUtils.java

```diff
@@ -65,9 +65,7 @@
 			return new byte[0];
 		}
 
-		ByteArrayOutputStream out = new ByteArrayOutputStream(BUFFER_SIZE);
-		copy(in, out);
-		return out.toByteArray();
+		return in.readAllBytes();
```

Comment:
This changes the behavior of this method. Formally, a maximum of `BUFFER_SIZE` (4096) was read, whereas `InputStream::readAllBytes()` reads `Integer.MAX_VALUE`.

Reply:
`BUFFER_SIZE` looks only for buffer. After all, all of bytes from `InputStream` will be read. `copy` method is really copy all bytes from `InputStream` to `OutputStream`

If replaced with `InputStream::readNBytes(BUFFER_SIZE)` then only `BUFFER_SIZE` size of all bytes will be read.